src/common/components/home/FeaturedProfilesSection.tsx

```diff
@@ -1,58 +1,27 @@
-import { useCallback, useEffect, useState } from 'react';
+import { useCallback, useEffect, useMemo, useState } from 'react';
 import { HomeSection, HomeSectionCarousel } from 'pages/index';
-import { ProfilePreviewData } from '@/types/types';
-import {
-  useFeaturedProfilesQuery,
-} from 'src/graphql/indexerTypes';
-import {
-  useWallet,
-  WalletContextState,
-} from '@solana/wallet-adapter-react';
 import ProfilePreview from '../elements/ProfilePreviewCard';
+import { QueryContext, useFeaturedProfiles } from '@/common/hooks/home';
 
 const CAROUSEL_ROWS: number = 2;
 const CAROUSEL_COLS: number = 3;
-const CAROUSEL_PAGES: number = 2;
-const N_LISTINGS: number = CAROUSEL_ROWS * CAROUSEL_COLS * CAROUSEL_PAGES;
-
-//TODO remove once other profiles have enough followers to preclude this one in the backend
-const DISALLOWED_PROFILES: string[] = ['ho1aVYd4TDWCi1pMqFvboPPc3J13e4LgWkWzGJpPJty'];
 
 export default function FeaturedProfilesSection(): JSX.Element {
-  const wallet: WalletContextState = useWallet();
-  // initial value hack to get loading card
-  // TODO pass in a loading boolean to bypass loading until the address is known
-  const [featuredProfiles, setFeaturedProfiles] = useState<ProfilePreviewData[]>(
-    [...Array(N_LISTINGS)].map((_) => ({ address: '', profile: {}, nftCounts: {} }))
-  );
-
-  const dataQuery = useFeaturedProfilesQuery({
-    variables: {
-      userWallet: wallet?.publicKey,
-      limit: CAROUSEL_PAGES * CAROUSEL_COLS * CAROUSEL_ROWS,
-    },
-  });
-
-  // get featured profiles once on page load and anytime later when the wallet has been (dis)connected
-  // by making the wallet pubkey one of the dependencies
-  useEffect(() => {
-    if (dataQuery.data?.followWallets && dataQuery.data.followWallets.length > 0) {
-      const profilesToShow = (dataQuery.data.followWallets as ProfilePreviewData[]).filter(
-        (p) => !DISALLOWED_PROFILES.includes(p.address)
-      );
-      setFeaturedProfiles(profilesToShow);
-    }
-  }, [wallet?.publicKey, setFeaturedProfiles, dataQuery.data?.followWallets]);
+  const queryContext: QueryContext<string[]> = useFeaturedProfiles();
+  const [excludedProfiles, setExcludedProfiles] = useState<string[]>([]);
 
   // when the server returns a profile with insufficient data to display the
   //  preview, remove it from the carousel
   const onInsufficientDataForAProfile = useCallback<(profileAddress: string) => void>(
     (profileAddress) => {
-      setFeaturedProfiles(featuredProfiles.filter((p) => p.address !== profileAddress));
+      excludedProfiles.push(profileAddress);
+      setExcludedProfiles(excludedProfiles);
     },
-    [featuredProfiles]
+    [excludedProfiles]
   );
 
+  const featuredProfiles: string[] = (queryContext.data ?? []).filter(a => !excludedProfiles.includes(a));
+
```

Comment:
Shouldnt we do the exclude at the api level so the limit still works. If you request 10 from the api but filter out 1 on the client you are going to have 9 objects.

Reply:
I think this particular filtering should happen on the client side, though that could feed back into the query. The client needs to determine what data are required to display a profile card, whereas the backend needs to serve whatever data are requested.

I could add optional filters to every API to exclude specific keys - in this case wallet addresses excluded from follow_wallets - and let the frontend update the query after determining a profile isnt worth showing.